libraries/AP_Frsky_Telem/AP_Frsky_Telem.cpp

```diff
@@ -445,7 +442,11 @@ void AP_Frsky_Telem::send_D(void)
     // send frame2 every second
     if (now - _D.last_1000ms_frame >= 1000) {
         _D.last_1000ms_frame = now;
-        send_uint16(DATA_ID_GPS_COURS_BP, (uint16_t)((_ahrs.yaw_sensor / 100) % 360)); // send heading in degree based on AHRS and not GPS
+        {
+            AP_AHRS &_ahrs = AP::ahrs();
+            WITH_SEMAPHORE(_ahrs.get_semaphore());
+            send_uint16(DATA_ID_GPS_COURS_BP, (uint16_t)((_ahrs.yaw_sensor / 100) % 360)); // send heading in degree based on AHRS and not GPS
```

Comment:
You can't hold the semaphore while doing this.

Reply:
how's it different from this?
https://github.com/ArduPilot/ardupilot/blob/73e51c480ad303542baf6aea27c5e4c124d17d7c/libraries/AP_OSD/AP_OSD_Screen.cpp#L1079-L1086

Reply:
I see...I'm accessing the serial port in send_uint16?

Reply:
> I see...I'm accessing the serial port in send_uint16?

Yep.  Usual story with semaphores - get in, do what you need to, get out.

The OSD case *does* really look suspect - I think those write() calls on the backends will all write into some sort of buffer so will be OK, but there's no guarantee on that.

Reply:
thanks Peter, I did fix that one.
I tried to be as "atomic" as I could.